lib/rest-api.php

```diff
@@ -265,6 +265,44 @@ function gutenberg_add_block_format_to_post_content( $response, $post, $request
 	return $response;
 }
 
+/**
+ * Include target schema attributes to links, based on whether the user can.
+ *
+ * @param WP_REST_Response $response WP REST API response of a post.
+ * @param WP_Post          $post The post being returned.
+ * @param WP_REST_Request  $request WP REST API request.
+ * @return WP_REST_Response Response containing the new links.
+ */
+function gutenberg_add_target_schema_to_links( $response, $post, $request ) {
+	$new_links  = array();
+	$orig_links = $response->get_links();
+	$post_type  = get_post_type_object( $post->post_type );
+	// Only Posts can be sticky.
+	if ( 'post' === $post->post_type ) {
+		if ( current_user_can( $post_type->cap->edit_others_posts )
+			&& current_user_can( $post_type->cap->publish_posts ) ) {
+			$new_links['https://api.w.org/action-sticky'] = array(
+				array(
+					'title'        => __( 'Sticky Post', 'gutenberg' ),
+					'href'         => $orig_links['self'][0]['href'],
+					'targetSchema' => array(
+						'type'       => 'object',
+						'properties' => array(
+							'sticky' => array(
+								'type'        => 'boolean',
+								'description' => __( 'Whether or not the object should be treated as sticky.', 'gutenberg' ),
+							),
+						),
+					),
+				),
```

Comment:
We have this issue all over the API, but this is a lot of data to be piling into the response just to indicate whether the UI can take some action. Do we really want to be repeating all of this:
![image](https://user-images.githubusercontent.com/442115/39499664-69b7e03a-4d7d-11e8-89c7-4b9e6eb6f88e.png)
for each post every time we're in a collection where we might _possibly_ want to sticky something? Our `_links` object is getting quite heavy.

I don't see a better option so I'm :+1: on the implementation, but I think we'll need to come back to that at some point.

Reply:
If we ever land generating a URL to access a schema, the `targetSchema` section could be replaced with a `$ref` for the property.

Reply:
We could also remove the `description` property but it'd make the link less helpful for clients.